src/app/containers/manuscript/manuscript-editor.tsx

```diff
@@ -88,13 +88,19 @@ export const ManuscriptEditor: React.FC = () => {
     [dispatch]
   );
 
-  const handleKeywordDelete = (keywordGroup: string, index: number): void => {
-    dispatch(manuscriptActions.deleteKeywordAction({ keywordGroup, index }));
-  };
+  const handleKeywordDelete = useCallback(
```

Comment:
I notice that if I delete a keyword from the middle of group, and then press undo the deleted keyword is restored but in the wrong location, it gets added to the beginning of the list.

Reply:
I couldn't reproduce this. Maybe there were other changes made before that combined with deleting produced this effect?